lib/awful/wibar.lua

```diff
@@ -228,22 +349,99 @@ local function set_position(wb, position, skip_reattach)
     wb:emit_signal("property::position", position)
 end
 
-local function get_stretch(w)
+function awfulwibar.get_stretch(w)
     return w._stretch
 end
 
-local function set_stretch(w, value)
+function awfulwibar.set_stretch(w, value)
     w._stretch = value
 
     attach(w, w.position)
 
     w:emit_signal("property::stretch", value)
 end
 
+
+function awfulwibar.get_update_workarea(w)
+    return w._private.update_workarea
+end
+
+function awfulwibar.set_update_workarea(w, value)
+    w._private.update_workarea = value
+
+    attach(w, w.position)
+
+    w:emit_signal("property::update_workarea", value)
+end
+
+
+function awfulwibar.set_margins(w, value)
+    if type(value) == "number" then
+        value = {
+            top = value,
+            bottom = value,
+            right = value,
+            left = value,
+        }
+    end
+
+   value = gtable.crush({
+        left   = 0,
+        right  = 0,
+        top    = 0,
+        bottom = 0
+    }, value or {}, true)
```

Comment:
Oh yes, I had it wrong in my review... I had to get back at it to see you already fixed it. 👍